File: WebDating/Entities/ProfileEntities/DatingProfile.cs
using System.ComponentModel.DataAnnotations;
using WebDating.Entities.UserEntities;

namespace WebDating.Entities.ProfileEntities
{
    public class DatingProfile
    {
        [Key]
        public int Id { get; set; }
        public int UserId { get; set; }
        public Gender DatingObject { get; set; }
        public Provice WhereToDate { get; set; }
        public int DatingAgeFrom { get; set; }
        public int DatingAgeTo { get; set; }
        public int HeightFrom { get; set; }
        public int HeightTo { get; set; }
        public int WeightFrom {  get; set; }
        public int WeightTo { get; set; }
        public IEnumerable<UserInterest> UserInterests { get; set; }
        public IEnumerable<Occupations> Occupations { get; set; }
        public AppUser User { get; set; }
    }
}

File: Client/client/src/app/_models/DatingProfile.ts
export interface EItem {
  value: number;
  displayName: string;
  interestType: number;
}
export interface EItem1 {
  value: number;
  displayName: string;
  occupationType: number;
}

export interface UserInterest {
  id: number;
  datingProfileId: number;
  interestName: number | string;
  interestNameCode: number;
  interestType: number;
}

export interface UserOccupation {
  id: number;
  datingProfileId: number;
  occupationName: number | string;
  occupationNameCode: number;
  occupationType: number;
}


export interface DatingProfile {
  id: number;
  userId: number;
  datingObject: string;
  datingObjectCode: number;
  height: string;
  heightCode: number;
  whereToDate: string;
  whereToDateCode: number;
  datingAgeFrom: number;
  datingAgeTo: number;
  weightFrom: number;
  weightTo: number;
  heightTo: number;
  heightFrom: number;
  userInterests: UserInterest[];
  occupations: UserOccupation[];
}

export interface DatingResponse {
  id: number;
  senderId: number;
  senderName: string;
  crushId: number;
  crushName: string;
  startDate: string;
  datingTimeSeconds: number;
  senderAvatar?: string;
  crushAvatar?: string;
}
